sktime/classification/distance_based/_shape_dtw.py

```diff
@@ -0,0 +1,477 @@
+                                             = scaling_factor_hog1d (default=0.1)
+
+            - 'compound'            : use a combination of two shape
+                                      descriptors simultaneously.
+                                    - params = weighting_factor
+                                              (default=None)
+                                               Defines how to scale
+                                               values of a shape
+                                               descriptor.
+                                               If a value is not given,
+                                               this value is tuned
+                                               by 10-fold cross-validation
+                                               on the training data.
+
+
+        shape_descriptor_functions  : string list, only applicable when the
+                                      shape_descriptor_function is
+                                      set to 'compound'.
+                                      Use a list of shape descriptor
+                                      functions at the same time.
+                                      (default = ['raw','derivative'])
+
+        metric_params               : dictionary for metric parameters
+                                      (default = None).
+        """
+        self.n_neighbors = n_neighbors
+        self.subsequence_length = subsequence_length
+        self.shape_descriptor_function = shape_descriptor_function
+        self.shape_descriptor_functions = shape_descriptor_functions
+        self.metric_params = metric_params
+        super(ShapeDTW, self).__init__()
+
+    def fit(self, X, y):
+        """
+        Method to perform training on the classifier.
+
+        Parameters
+        ----------
+        X - pandas dataframe of training data of shape [num_insts,1].
+        y - list of class labels of shape [num_insts].
+
+        Returns
+        -------
+        self : the shapeDTW object
+        """
+        # Perform preprocessing on params.
+        if not (isinstance(self.shape_descriptor_function, str)):
+            raise TypeError("shape_descriptor_function must be an 'str'. \
+                            Found '" +
+                            type(self.shape_descriptor_function).__name__ +
+                            "' instead.")
+
+        X, y = check_X_y(X, y, enforce_univariate=False)
+
+        if self.metric_params is None:
+            self.metric_params = {}
+
+        # If the shape descriptor is 'compound',
+        # calculate the appropriate weighting_factor
+        if self.shape_descriptor_function == "compound":
+            self._calculate_weighting_factor_value(X, y)
+
+        self.trainData = X
+        self.trainDataClasses = y
+
+        # get the number of attributes and instances
+        num_atts = self.trainData.iloc[0, 0].shape[0]
```

Comment:
could we rename `num_atts` to `n_timepoints`, is the length of the series, isn't it?